estate/models/estate_property.py

```diff
@@ -20,39 +22,70 @@ class EstateProperty(models.Model):
         ('east', 'East'),
         ('west', 'West')
     ])
+    status = fields.Selection([
+        ('new', 'New'),
+        ('sold', 'Sold'),
+        ('canceled', 'Canceled')], default='new', required=True)
     tag_ids = fields.Many2many(
-        'estate.property.tag', 
+        'estate.property.tag'
     )
     salesman_id = fields.Many2one(
         'res.users', 
         default=lambda self: self.env.user
     )
-    
     buyer_id = fields.Many2one(
         'res.partner', 
         copy=False
     )
-    
     offer_ids = fields.One2many(
     'estate.property.offer',
-    'property_id')
-
+    'property_id',
+    cascade=True
+    )
+    total_area = fields.Integer(compute="_compute_total_area", store=True)
+    best_offer = fields.Float(compute="_compute_best_price", store=True)
+    
     @api.depends('area', 'garden_area')
     def _compute_total_area(self):
         for record in self:
             record.total_area = record.area + record.garden_area
-
-    total_area = fields.Integer(compute="_compute_total_area", store=True)
-        
+       
     @api.depends('offer_ids.price')
     def _compute_best_price(self):
         for record in self:
             record.best_offer = max(record.offer_ids.mapped('price'), default=0)
-
-    best_offer = fields.Float(compute="_compute_best_price", store=True)
     
     @api.onchange('garden')
     def _onchange_garden(self):
         if self.garden:
             self.garden_area = 10
             self.garden_orientation = 'north'
+            
+    def cancelRequest(self):
```

Comment:
```suggestion
    def cancel_request(self):
```
name methods in snake case